frontend/src/components/courses/UnitSidebar.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState } from "react";
+import {
+  Box,
+  Button,
+  Divider,
+  Drawer,
+  List,
+  ListItemButton,
+  ListItemText,
+  useTheme,
+} from "@mui/material";
+import ArrowBackIosIcon from "@mui/icons-material/ArrowBackIos";
+import { CourseUnit } from "../../types/CourseTypes";
+import { neutral } from "../../theme/palette";
+
+interface UnitSideBarProps {
+  courseUnits: CourseUnit[];
+  handleClose: () => void;
+  open: boolean;
+}
+
+export default function UnitSidebar(props: UnitSideBarProps) {
+  const theme = useTheme();
+  const { courseUnits, handleClose, open } = props;
+
+  const [selectedIndex, setSelectedIndex] = useState(0);
+
+  const handleListItemClick = (
+    event: React.MouseEvent<HTMLDivElement, MouseEvent>,
+    index: number,
+  ) => {
+    setSelectedIndex(index);
+  };
+
+  return (
+    <Drawer
+      sx={{
+        width: "20%",
+        flexShrink: 0,
+        "& .MuiDrawer-paper": {
+          width: "20%",
+          boxSizing: "border-box",
+        },
+        display: open ? "block" : "none",
+      }}
+      variant="persistent"
+      anchor="left"
+      open={open}
+    >
+      <Box height="100%" sx={{ backgroundColor: theme.palette.neutral.light }}>
+        <Box
+          height="59px"
+          display="flex"
+          p="12px"
+          alignItems="center"
+          justifyContent="space-between"
+          fontWeight="700"
+          fontSize="16px"
+        >
+          Units
+          <Button
+            type="button"
+            sx={{
+              p: "8px",
+              fontSize: "12px",
+            }}
+            endIcon={<ArrowBackIosIcon />}
+            onClick={handleClose}
+          >
+            Close
+          </Button>
+        </Box>
+        <List sx={{ width: "100%" }}>
+          {courseUnits.map((course, index) => {
+            return (
+              <React.Fragment key={course.id}>
+                <ListItemButton
+                  sx={{
+                    py: "15px",
+                    px: "32px",
+                    backgroundColor:
+                      selectedIndex === index ? neutral[90] : "transparent",
+                  }}
+                  onClick={(event) => handleListItemClick(event, index)}
+                >
+                  <ListItemText
+                    disableTypography
+                    primary={`${course.displayIndex}. ${course.title}`}
+                    sx={
+                      selectedIndex === index
+                        ? theme.typography.titleMedium
+                        : theme.typography.bodyLarge
+                    }
+                  />
+                </ListItemButton>
+                {index !== courseUnits.length - 1 && (
+                  <Divider component="li" sx={{ color: "#FCF8F8" }} />
+                )}
+              </React.Fragment>
```

Comment:
This is creating an accessibility issue (according to the accessibility checker I ran) - [link](https://dequeuniversity.com/rules/axe/4.9/list?application=axeAPI). TLDR: the immediate child of a `<ul>` should be `<li>`, not something else (e.g. `<div>`) - this is so that screenreaders can interpret lists correctly. 

The `React.Fragment` seems to be causing this issue. Using `ListItem` will get us the `<li>` tags we require, so we should wrap each item in our list with it. When I did this, the Divider disappeared... Alternate proposal: use bottom border to illustrate a divider (this is actually what was in the design when I dev moded it -  so this is not a janky alternative :))

So my suggestion would be to delete the `React.Fragment` and `Divider` part, then wrap the `ListItemButton` with:
```
<ListItem
  key={course.id}
  disablePadding
  sx={{
    borderBottom: 1,
    borderColor:
      index !== courseUnits.length - 1
        ? "#DBE4E7"
        : "transparent",
  }}
>
```